src/net/socket.rs

```diff
@@ -462,4 +468,73 @@ mod tests {
             .collect();
         assert_eq!(sent_events, vec![0, 1, 35]);
     }
+
+    #[test]
+    fn really_bad_network_keeps_chugging_along() {
```

Comment:
This starts to look like an integration test. Maybe we want to move those to /tests, I am fine to have it here as well. But just a thought.